app/src/main/java/com/architectcoders/arquitectomarvel/ui/main/MainActivity.kt

```diff
@@ -1,72 +1,54 @@
 package com.architectcoders.arquitectomarvel.ui.main
 
-import android.content.Intent
 import android.os.Bundle
-import android.view.View
 import androidx.appcompat.app.AppCompatActivity
-import androidx.core.app.ActivityOptionsCompat
 import androidx.core.view.isVisible
+import androidx.lifecycle.Observer
 import com.architectcoders.arquitectomarvel.R
 import com.architectcoders.arquitectomarvel.databinding.ActivityMainBinding
-import com.architectcoders.arquitectomarvel.model.EXTRA_SELECTED_HERO
-import com.architectcoders.arquitectomarvel.model.Repository
-import com.architectcoders.arquitectomarvel.model.autoFitColumnsForGridLayout
-import com.architectcoders.arquitectomarvel.model.characters.Result
-import com.architectcoders.arquitectomarvel.model.toast
+import com.architectcoders.arquitectomarvel.model.*
 import com.architectcoders.arquitectomarvel.ui.detail.HeroDetailActivity
-import timber.log.Timber
 
-class MainActivity : AppCompatActivity(), MainPresenter.View {
+class MainActivity : AppCompatActivity() {
 
     private lateinit var binding: ActivityMainBinding
-    private val presenter by lazy { MainPresenter(Repository(this)) }
-    private val adapterList by lazy { AdapterList(presenter::onResultClick) }
+    private lateinit var viewModel: MainViewModel
+    private lateinit var adapter: AdapterList
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        binding = ActivityMainBinding.inflate(layoutInflater)
-        setContentView(binding.root)
-        presenter.onCreate(this)
+        initBinding()
+        initViewModel()
     }
 
-    override fun initViews() {
+    private fun initBinding() {
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
         binding.mainHeroList.autoFitColumnsForGridLayout(resources.getDimension(R.dimen.avatar_width))
-        binding.mainHeroList.adapter = adapterList
     }
+    private fun initViewModel(){
+        viewModel = getViewModel { MainViewModel(Repository(application)) }
+        adapter = AdapterList(viewModel::onResultClick)
+        binding.mainHeroList.adapter = adapter
+        viewModel.model.observe(this, Observer(::updateUI))
+        viewModel.navigation.observe(this, Observer { event ->
+            event.getContentIfNotHandled()?.let {
 
-    override fun showProgress() {
-        binding.progress.isVisible = true
-    }
+                toast("Hero selected ${it.name}")
+                //TODO: putExtra Result: it
+                //startActivity<HeroDetailActivity> {}
```

Comment:
@jujoru you forgot to uncomment this line to go to the detail. Could you add the option parameter like I did in the deleted function "navigateTo"?

Reply:
I have left it commented, even knowing that it is not advisable to leave "TO DO" ​​comments when committing, for when MVVM is implemented in the Details. Anyway I have checked, and in the Detail the getParcelable is already prepared, this is totally independent of the MVVM implementation, therefore I am going to complete it.